lib/emerald/parser.rb

```diff
@@ -34,8 +36,13 @@ def parse_whitespace(source)
       [nil, rest_of_source]
     end
 
-    def parse_atom(source)
-      pattern = /\A[a-z\-]+/
+    def parse_atom(source, non_whitespace_char)
+      case non_whitespace_char
+      when /\A[a-zA-Z\-]+/
+        pattern = /\A[a-zA-Z\-]+/
+      when /\d+/
+        pattern = /\d+/
+      end
       atom_value = pattern.match(source).to_s
```

Comment:
I feel like this could be done in a nicer way, instead of another switch statement. Any suggestions?